src/main/kotlin/uk/gov/justice/digital/hmpps/sendlegalmailtoprisonsapi/onetimecode/OneTimeCodeAttemptsRepository.kt

```diff
@@ -0,0 +1,16 @@
+package uk.gov.justice.digital.hmpps.sendlegalmailtoprisonsapi.onetimecode
+
+import org.springframework.data.annotation.Id
+import org.springframework.data.redis.core.RedisHash
+import org.springframework.data.repository.CrudRepository
+import org.springframework.stereotype.Repository
+
+@Repository
+interface OneTimeCodeAttemptsRepository : CrudRepository<OneTimeCodeAttempts, String>
+
+@RedisHash(value = "oneTimeCodeAttempts")
+data class OneTimeCodeAttempts(
+  @Id
+  val sessionId: String,
+  var attempts: Set<String>?,
+)
```

Comment:
We now have an extra Redis cache to capture all attempts at the one time code which is required so that we don't reset the TTL on the main one time code cache. We use a set due to a bizarre requirement - allow up to 6 attempts but ignore duplicate failed attempts 🤷

Reply:
> a bizarre requirement
lol 🤣